components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeEndpoint.java

```diff
@@ -16,15 +16,27 @@
  */
 package org.apache.camel.component.xchange;
 
+import static org.apache.camel.component.xchange.XChangeConfiguration.SERVICE_MARKETDATA;
+import static org.apache.camel.component.xchange.XChangeConfiguration.SERVICE_METADATA;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
 import org.apache.camel.Consumer;
 import org.apache.camel.Processor;
 import org.apache.camel.Producer;
-import org.apache.camel.impl.DefaultPollingEndpoint;
+import org.apache.camel.impl.DefaultEndpoint;
 import org.apache.camel.spi.UriEndpoint;
 import org.apache.camel.spi.UriParam;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.meta.CurrencyMetaData;
+import org.knowm.xchange.dto.meta.CurrencyPairMetaData;
+import org.knowm.xchange.dto.meta.ExchangeMetaData;
+import org.knowm.xchange.utils.Assert;
 
-@UriEndpoint(firstVersion = "2.21.0", scheme = "xchange", title = "XChange", syntax = "xchange:name", consumerClass = XChangeConsumer.class, label = "api")
-public class XChangeEndpoint extends DefaultPollingEndpoint {
+@UriEndpoint(firstVersion = "2.21.0", scheme = "xchange", title = "XChange", syntax = "xchange:name", label = "api")
```

Comment:
You need to set consumerOnly = true, also label = "api" should be label = "currency" or "crypto-currency" or something that is a good label for this kind of components (think label = group)

Reply:
How about blockchain?